### ISSUE TITLE
security/libsodium: used -fstack-protector, didn't link libssp

### ISSUE THREAD

**Issue by nshalman** [TritonDataCenter/pkgsrc-legacy#209]
with 2014Q2 I'm still getting an error with libsodium when I try to load it with python's ctypes:
OSError: ld.so.1: python2.7: fatal: relocation error: file /opt/local/lib/libsodium.so.4.5.0: symbol __stack_chk_guard: referenced symbol not found

@richlowe says "used -fstack-protector, didn't link libssp"
@jperkin says "and what richlowe said, though I don't really understand how that works as gcc specs explicitly add lssp if -fstack-protector is used afair"

**Comment by nshalman** [comment]
Steps to recreate the problem (tested in Manta):

```
cd /
mv /opt/local /opt/old
curl http://pkgsrc.joyent.com/packages/SmartOS/bootstrap/bootstrap-2014Q2-x86_64.tar.gz | gtar xz
pkgin up
pkgin -y in python27 libsodium
python2.7 -c 'import ctypes ; print ctypes.cdll.LoadLibrary("/opt/local/lib/libsodium.so.4.5.0")'
```

**Comment by nshalman** [comment]
Fixed in 138e650de5010a3691162e8022841e063546a2e0 by @jperkin.
Thanks!!
